pages/form-builder/publish.tsx

```diff
@@ -1,13 +1,45 @@
-import React, { ReactElement } from "react";
+import React, { ReactElement, useEffect } from "react";
 import { useTranslation } from "next-i18next";
+import { useRouter } from "next/router";
 import { NextPageWithLayout } from "../_app";
 import { PageProps } from "@lib/types";
 import { getServerSideProps } from "./index";
 import { Publish, Template, PageTemplate } from "@components/form-builder/app";
+import { useTemplateStore } from "@formbuilder/store";
 
 const Page: NextPageWithLayout<PageProps> = () => {
   const { t } = useTranslation("form-builder");
   const title = `${t("gcFormsPublish")} — ${t("gcForms")}`;
+  const router = useRouter();
+
+  const { id, isPublished } = useTemplateStore((s) => ({
+    id: s.id,
+    isPublished: s.isPublished,
+  }));
+
+  useEffect(() => {
```

Comment:
Logic added here in case we hit the /publish route direct and not already handled by the server